globalmangrovewatch/gmw-users
# 7.3/7.4/7.5 & Customize tabular sections for each form

- 7.3 (`funderName`,` funderType`, `percentage`)
- 7.4 monetary inputs
- 7.5 cost breakdowns
- for the select, the framework review doc did not specify to clarify 'other', so I intentionally left an 'other' input option out

**Customizing tabular sections for each form**

At first I tried to make a generic `AddTabularInputRow` and `TabularInputRow` components that I thought I could reuse in other tabular input sections. As I am now on the third tabular question type, I realized this was a bit of a losing strategy since each of them are all quite different from each other. I decided customizing these sections for each form component was a simpler solution. For example, some sections have unique logic, question types, and the amount of columns. Some sections with more columns required a vertical layout vs horizontal, etc.

**To test:**
1. go to Costs Form section
2. Select 'Monetary' in 7.1 (required logic for 7.3g to appear)
3. scroll to 7.3
4. try adding an input row, deleting, editing, saving the form, and refreshing
5. test out 7.4 
6. test out 7.5

- since I also made naming adjustments in the two other tabular input row sections from different forms, its worth testing those out too incase I missed something during my testing. 

  This includes: 
  Pre Restoration Assessment, 5.3g
  Site Interventions, 6.2c
